src/query/query_list.h

```diff
@@ -169,63 +172,54 @@ struct xasl_node_header
 typedef struct xasl_id XASL_ID;
 struct xasl_id
 {
-  VPID first_vpid;		/* first page real identifier */
-  VFID temp_vfid;		/* temp file volume and file identifier */
+  SHA1Hash sha1;		/* SHA-1 hash generated from query string. */
+  volatile INT32 cache_flag;	/* Multiple-purpose field used to handle XASL cache. */
   CACHE_TIME time_stored;	/* when this XASL plan stored */
 };				/* XASL plan file identifier */
 
 #define XASL_ID_SET_NULL(X) \
   do \
     { \
-      (X)->first_vpid.pageid = NULL_PAGEID; \
-      (X)->first_vpid.volid  = NULL_VOLID; \
-      (X)->temp_vfid.fileid = NULL_FILEID; \
-      (X)->temp_vfid.volid = NULL_VOLID; \
-      /*CACHE_TIME_RESET(&((X)->time_stored))*/ \
+      (X)->sha1.h[0] = 0; \
+      (X)->sha1.h[1] = 0; \
+      (X)->sha1.h[2] = 0; \
+      (X)->sha1.h[3] = 0; \
+      (X)->sha1.h[4] = 0; \
+      (X)->cache_flag = 0; \
       (X)->time_stored.sec = 0; \
       (X)->time_stored.usec = 0; \
     } \
   while (0)
 
-#define XASL_ID_IS_NULL(X) (((XASL_ID *) (X) != NULL) && (X)->first_vpid.pageid == NULL_PAGEID)
+#define XASL_ID_IS_NULL(X) (((XASL_ID *) (X) != NULL) && (X)->time_stored.sec == 0)
 
 #define XASL_ID_COPY(X1, X2) \
-  *(X1) = *(X2)
-
-#define MAKE_PSEUDO_XASL_ID_FROM_BTREE(X,B) \
   do \
     { \
-      (X)->first_vpid.pageid = (B)->root_pageid; \
-      (X)->first_vpid.volid  = 0x8000 | (B)->vfid.volid; \
-      (X)->temp_vfid.fileid = (B)->vfid.fileid; \
-      (X)->temp_vfid.volid = 0x8000 | (B)->vfid.volid; \
-      (X)->time_stored.sec = 0; \
-      (X)->time_stored.usec = 0; \
+      (X1)->sha1 = (X2)->sha1; \
+      (X1)->time_stored = (X2)->time_stored; \
+      /* Do not copy cache_flag. */ \
     } \
   while (0)
 
 /* do not compare with X.time_stored */
 #define XASL_ID_EQ(X1, X2) \
     ((X1) == (X2) \
-     || (((X1)->first_vpid.pageid == (X2)->first_vpid.pageid && (X1)->first_vpid.volid == (X2)->first_vpid.volid) \
-         && ((X1)->temp_vfid.fileid == (X2)->temp_vfid.fileid && (X1)->temp_vfid.volid == (X2)->temp_vfid.volid)))
+     || (SHA1Compare (&(X1)->sha1, &(X2)->sha1) == 0 \
+         && (X1)->time_stored.sec == (X2)->time_stored.sec \
+         && (X1)->time_stored.usec == (X2)->time_stored.usec))
 
-#define OR_XASL_ID_SIZE (OR_LOID_SIZE + OR_CACHE_TIME_SIZE)
+#define OR_XASL_ID_SIZE (OR_SHA1_SIZE + OR_CACHE_TIME_SIZE)
 
 /* pack XASL file id (XASL_ID)
      - borrow LOID structure only for transmission purpose
```

Comment:
@ribram 

You can delete LOID related comments from `OR_PACK_XASL_ID` and `OR_UNPACK_XASL_ID`.